jerry-main/main-unix-snapshot.c

```diff
@@ -0,0 +1,308 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+
+#include "jerryscript.h"
+#include "jerryscript-port.h"
+
+#include "cli.h"
+
+/**
+ * Maximum size for loaded snapshots
+ */
+#define JERRY_BUFFER_SIZE (1048576)
+
+/**
+ * Standalone Jerry exit codes
+ */
+#define JERRY_STANDALONE_EXIT_CODE_OK   (0)
+#define JERRY_STANDALONE_EXIT_CODE_FAIL (1)
+
+static uint8_t input_buffer[ JERRY_BUFFER_SIZE ];
+static uint32_t output_buffer[ JERRY_BUFFER_SIZE / 4 ];
+
+/**
+ * Loading a single file into the memory.
+ *
+ * @return true - if loading is successful
+ *         false - otherwise
+ */
+static size_t
+read_file (uint8_t *input_pos_p, /**< next position in the input buffer */
+           const char *file_name) /**< file name */
+{
+  FILE *file = fopen (file_name, "r");
+
+  if (file == NULL)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: failed to open file: %s\n", file_name);
+    return 0;
+  }
+
+  size_t max_size = (size_t) (input_buffer + JERRY_BUFFER_SIZE - input_pos_p);
+
+  size_t bytes_read = fread (input_pos_p, 1u, max_size, file);
+  fclose (file);
+
+  if (bytes_read == 0)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: failed to read file: %s\n", file_name);
+    return 0;
+  }
+
+  if (bytes_read >= max_size)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: file too large: %s\n", file_name);
+    return 0;
+  }
+
+  printf ("Input file '%s' (%d bytes) loaded.\n", file_name, (int) bytes_read);
```

Comment:
this is still a `printf`, while the rest of the diagnostics is via `jerry_port_log`

Reply:
There is no suitable type for this in port log.